src/garage/envs/base.py

```diff
@@ -148,7 +148,19 @@ def step(self, action):
                 debugging, and sometimes learning)
 
         """
-        return self.env.step(action)
+        observation, reward, done, info = self.env.step(action)
+        # gym envs that are wrapped in timelimit wrapper modify
+        # the done/termination signal to be true whenever a time
+        # limit expiration occurs. The following statement sets
+        # the done signal to be True only if caused by an
+        # environment termination, and not a time limit
+        # termination. The time limit termination signal
+        # will be saved inside env_infos as
+        # 'TimeLimitTerminated'
+        if 'TimeLimit.truncated' in info:
+            info['TimeLimitTerminated'] = done
+            done = not info['TimeLimit.truncated']
```

Comment:
If `TimeLimit.truncated` is ever set to `False`, then this would be wrong. Please change the logic so it's correct in that case.

Reply:
I don't think this is the case. Looking at the code of the time limit wrapper, it seems that TimeLimit.truncated equals the Not of the original done signal caused by any environment termination. Is there something I'm missing here @krzentner ?

Reply:
I don't think this is the case. Looking at the code of the time limit wrapper, it seems that TimeLimit.truncated equals the Not of the original done signal caused by any environment termination. Is there something I'm missing here @krzentner ?

Reply:
can you provide a link?

Reply:
For those following along at home: https://github.com/openai/gym/blob/master/gym/wrappers/time_limit.py#L19

I agree that the logic is weird if an environment reached both termination and the time limit in the same step, but due to the implementation of `TimeLimit`, `not info[TimeLimit.truncated]` is the only way to recover the original `done` signal.

Reply:
Oh yes, thanks, that is the link. It is the reference to the misunderstanding about `info[TimeLimit.truncated] ` @krzentner

Reply:
Okay, I guess that makes sense (even if it's weird).

Reply:
Want to clarify -- does `GarageEnv.TimeLimitTerminated` tell us it reached time limit or not and now `done` will only be true if it's from env termination?